.vsts.pipelines/jobs/collect-artifacts.yml

```diff
@@ -0,0 +1,104 @@
+jobs:
+- job: Collect_Artifacts
+
+  dependsOn: ${{ parameters.dependsOn }}
+
+  displayName: Collect Artifacts
+
+  pool: ${{ parameters.pool }}
+
+  steps:
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download CentOS artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline
+        downloadPath: '$(Build.StagingDirectory)/centos'
```

Comment:
IMO you should use an [`each`](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#iterative-insertion) and pass a list of objects to this template to keep this out of the implementation yaml. You can get this down to only have a a single input list of objects that you reprocess to create the `dependsOn` and the sequence of download steps. The overall point being to keep the platform listings in a single file.